pkg/controller/migplan/registry_health_failed.go

```diff
@@ -0,0 +1,129 @@
+package migplan
+
+import (
+	"github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"time"
+)
+
+// A registry health checker which routinely checks whether registries are healthy or not for migplan's failed migrations
+type registryHealthFailed struct {
+	hostClient client.Client
+	Interval   time.Duration
+	handler    handler.EventHandler
+	queue      workqueue.RateLimitingInterface
+	predicates []predicate.Predicate
+	podLabels  map[string]string
+}
+
+// Start the health checks
+func (r *registryHealthFailed) Start(
+	handler handler.EventHandler,
+	queue workqueue.RateLimitingInterface,
+	predicates ...predicate.Predicate) error {
+
+	r.handler = handler
+	r.queue = queue
+	r.predicates = predicates
+	go r.run()
+
+	return nil
+}
+
+// Enqueue a reconcile request event for migplan
+func (r *registryHealthFailed) enqueue(plan v1alpha1.MigPlan) {
+	event := event.GenericEvent{
+		Meta:   &plan.ObjectMeta,
+		Object: &plan,
+	}
+	for _, predicate := range r.predicates {
+		if !predicate.Generic(event) {
+			return
+		}
+	}
+
+	r.handler.Generic(event, r.queue)
+}
+
+//Run the health checks for registry pods
+func (r *registryHealthFailed) run() {
+	//List all the migplans that have migrations in failed state using the hostClient
+	//Now using srcClient and destClient for each migplan find the registry pods, if registry container is not ready then enqueue this migplan
+	//repeat all the above steps for all the plans for both clusters
+
+	for {
+		time.Sleep(r.Interval)
+		planList, err := v1alpha1.ListFailedPlans(r.hostClient)
+		if err != nil {
+			log.Trace(err)
+			return
+		}
+		if planList != nil {
+			for _, plan := range planList {
+				//TODO avoid race condition check
+				srcCluster, err := plan.GetSourceCluster(r.hostClient)
+				if err != nil {
+					log.Trace(err)
+					//TODO Error condition would result in loosing the watch util the pod restarts
+					return
```

Comment:
we talked about removing the returns here so the go routine does not die

Reply:
Done